src/App.tsx

```diff
@@ -1,14 +1,99 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { TodoWithUser } from './types/TodoWithUser';
+import { FilterType } from './types/FilterType';
+import { ifInclude } from './components/IfInclude';
+
+async function loadTodos() {
+  const todosFromServer = await getTodos();
+
+  return todosFromServer;
+}
+
+async function loadUserForTodo(todo:Todo) {
+  const userFromServer = await getUser(todo.userId);
+
+  return userFromServer;
+}
 
 export const App: React.FC = () => {
+  const [isOpenedModal, setIsOpenedModal] = useState(false);
+  const [openedTodo, setOpenedTodo] = useState<TodoWithUser | null>(null);
+  const [isDataLoad, setIsDataLoaded] = useState(false);
+  const [isUserLoad, setIsUserLoaded] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosServer, setTodosServer] = useState<Todo[]>([]);
```

Comment:
instead of using several states, you can create a variable, for example `visibleTodos`, that will contain the results of the filter todos function
use `useMemo` hook and pass filter params into dependencies